lib/matplotlib/tests/test_axes.py

```diff
@@ -4946,6 +4946,11 @@ def test_ls_ds_conflict():
         plt.plot(range(32), linestyle='steps-pre:', drawstyle='steps-post')
 
 
+@image_comparison(baseline_images=['bar_uint8'], extensions=['png'])
```

Comment:
I've replaced the uint8 test with a numerical check that the patches end up in the right place. Can't think of an easy way to merge the rectangle with units, but could maybe just drop the image test and just check that a rectangle with units doesn't throw any errors?